deeplake/core/dataset/dataset.py

```diff
@@ -232,6 +233,9 @@ def __init__(
         ] = []  # This is a stack to support nested with contexts
         self._indexing_history: List[int] = []
 
+        for temp_tensor in self._temp_tensors:
```

Comment:
Wouldn't `self._temp_tensors` be always empty at this point??

Reply:
it will be populated when loading meta in `self.set_derived_attributes()`